### ISSUE TITLE
Actuator's "channels" endpoint may not properly expose binding channel information

### ISSUE THREAD

**Issue by yoshikawaa** [spring-cloud/spring-cloud-stream#2612]
### versions
* Spring Boot: 2.7.6
* Spring Cloud: 2021.0.5

### Problem
```yml
management:
  endpoints:
    web:
      exposure:
        include:
          - health # (1)
          - health-path
          - bindings
          - channels # (2)
```

If I expose the endpoints that Spring Cloud Stream provides to Spring Boot Actuator via HTTP, there is a problem that exposing the "health" endpoint prevents the "channels" endpoint from retrieving binding channel information.

* If "health" is not exposed, "channels" exposes binding channel information normally.
* If "health" is exposed, "channels" exposes empty information.

### Cause
This is because if the "health" is exposed, the processing timing of the `BindableFunctionProxyFactory` will be delayed, and the `ChannelsEndpoint` will not be initialized in time.

**Comment by olegz** [comment]
Can you please provide small example that reproduces the issue to speed up diagnostics and fix. You can push it to github and post a link or attach it as zip file

**Comment by yoshikawaa** [comment]
I pushed a minimal sample, please check.
https://github.com/yoshikawaa/spring-cloud-stream-2612-demo
* Run application -> include `health,channels` endpoints.
* Run junit tests -> both of `health,channels` and `channels` endpoints.

**Comment by bebers22** [comment]
Is there any plan to fix it for spring boot 2.7.8  with SCS 3.2.6 ?

**Comment by yoshikawaa** [comment]
@bebers22 
https://github.com/spring-cloud/spring-cloud-stream/commits/v3.2.7
GH-2612 seems to be fixed in 3.2.7 for 3.2.x.

https://github.com/spring-cloud/spring-cloud-release/blob/v2021.0.6/spring-cloud-dependencies/pom.xml#L32
Spring Cloud Stream 3.2.7 included in Spring Cloud 2021.0.6.
